homeassistant/components/neato/__init__.py

```diff
@@ -77,10 +70,20 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
     )
 
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+    try:
+        await session.async_ensure_token_valid()
+    except aiohttp.ClientResponseError as ex:
+        _LOGGER.debug("API error: %s (%s)", ex.code, ex.message)
+        if ex.code in (401, 403):
+            raise ConfigEntryAuthFailed("Token not valid, trigger renewal") from ex
```

Comment:
Just after merging this I realized that maybe we should raise `ConfigEntryNotReady` in the other case?

Reply:
Separate PR or addition to strict typing one ?

Reply:
Please do a separate PR, thanks!